compiler/aarch64/codegen/FPTreeEvaluator.cpp

```diff
@@ -453,10 +453,18 @@ static TR::Instruction *iffcmpHelper(TR::Node *node, TR::ARM64ConditionCode cc,
 
       TR::RegisterDependencyConditions *deps = generateRegisterDependencyConditions(cg, thirdChild, 0);
       result = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, dstLabel, cc, deps);
+      if (isUnordered)
+         {
+         result = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, dstLabel, TR::CC_VS, deps);
```

Comment:
You shouldn't use the same register dependency conditions on more than one instruction.  In this case, for unordered, I think it is fine to have the `deps` on just this instruction, and skip it on the previous instruction because there is nothing in between that will mess up the register state.  For unordered, however, you need to still add the `deps` to the original instruction since you won't be generating the unordered branch.

Reply:
I did the same as the 32-bit ARM code for unordered comparison.  I guess it also has to be modified, then:

https://github.com/eclipse/omr/blob/4beca561d294ed2a88099f43f5e8603e672996ff/compiler/arm/codegen/FPTreeEvaluator.cpp#L2253-L2257